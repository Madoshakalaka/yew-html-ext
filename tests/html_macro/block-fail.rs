use yew_html_ext::html;

fn compile_fail() {
    html! {
        <>
            { () }
        </>
    };

    let not_tree = || ();
    html! {
        <div>{ not_tree() }</div>
    };
    html! {
        <>{ for (0..3).map(|_| not_tree()) }</>
    };
}

fn main() {}
