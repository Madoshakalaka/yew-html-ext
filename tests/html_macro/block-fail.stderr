error[E0277]: the trait bound `(): Into<NodeSeq<(), VNode>>` is not satisfied
 --> tests/html_macro/block-fail.rs:6:15
  |
6 |             { () }
  |               ^^
  |               |
  |               the trait `std::fmt::Display` is not implemented for `()`
  |               required by a bound introduced by this call
  |
  = note: required for `()` to implement `ToString`
  = note: required for `VNode` to implement `From<()>`
  = note: required for `()` to implement `Into<VNode>`
  = note: 2 redundant requirements hidden
  = note: required for `()` to implement `Into<NodeSeq<(), VNode>>`

error[E0277]: the trait bound `(): Into<VNode>` is not satisfied
  --> tests/html_macro/block-fail.rs:15:17
   |
15 |         <>{ for (0..3).map(|_| not_tree()) }</>
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `std::fmt::Display` is not implemented for `()`
   |
   = note: required for `()` to implement `ToString`
   = note: required for `VNode` to implement `From<()>`
   = note: required for `()` to implement `Into<VNode>`
note: required by a bound in `into_node_iter`
  --> /home/maa/.cargo/git/checkouts/yew-07f42ef43f99cbca/69e3b5f/packages/yew/src/utils/mod.rs:11:8
   |
8  | pub fn into_node_iter<IT, T, R>(it: IT) -> impl Iterator<Item = R>
   |        -------------- required by a bound in this function
...
11 |     T: Into<R>,
   |        ^^^^^^^ required by this bound in `into_node_iter`
